### PR TITLE
Shuffled `destination-jdbc` into `bases` folder

### LINKED ISSUE #25070: One Java One Publish: Move `destination-jdbc` into `base-java`
## What
There is a tight coupling between the `destination-jdbc` components (`StagingConsumerFactory`, `JdbcConsumerFactory`) and the buffering components that exist in `base-java` (`BufferedStreamConsumer` and `BufferingStrategy`). This causes issues now that the buffer needs to be aware of _how_ buffered data is flushed and written to the destination storage in an async world

The goal of this change is to bundle the coupled components into the same package/folder _and then_ break the package into an `old` and `new` version which would be representative of `sequential` and `async` respectively

What this also unlocks is the ability for users to create a better interface for _how_ JDBC destination connectors should be built. The problems we see right now are that buffers are coupled with _how_ a destination connector loads data, which tables are created, deleted, truncated, and causes difficulty upgrading functionality without affecting other connectors. While this change will _not_ completely decouple the connector buffer/state management changes it will allow for an easier path to move JDBC connectors one at a time to async reading/writing

# How
1. Move `connectors:destination-jdbc` modules into `bases` folder
2. Version the aggregated `base-java` into `new`/`old` or `sequential`/`async`

## Definition of Done
1. All 16 connectors that use `destination-jdbc` should be able to pass integration tests (or if already failing to fail on same tests)

### PR DESCRIPTION
**Pull request by ryankfu**
## What
Closes airbytehq/airbyte#25070

Hoists the `destination-jdbc` "connector" into the `/bases` folder to be more representative of the code. It serves two purposes related to one Java one publish:
- Clears the way for cohesive base modules 
- Allows the newly separated `bases-destination-jdbc` to be versioned on a `sequential` write and `async` write folder

The separation is important because it unlocks the two priority EPICs for this quarter to not have conflicts but also unlocks further trial and error of building an interface similar to the python HTTP CDK for JDBC destination CDK

This should be seen as a no-op change, merely shifting files around and removing some unused files as well as renaming to make destination modules more clear

## How
1. Moves the `destination-jdbc` package into `/bases` folder
2. Renames `destination-jdbc` into `bases-destination-jdbc` (NOTE: base(S) because gradle executes the closures in lexicographical order and `base-java` and `base-normalization` is needed to execute _before_ `bases-destination-jdbc`)
3. Runs integration tests on all JDBC connectors where integration tests already pass to validate builds are still working
4. Removes an unused `Dockerfile` within `destination-jdbc` that wasn't being used publicly

## Recommended reading order
1. `build.gradle`
3. `settings.gradle`

## 🚨 User Impact 🚨
*Are there any breaking changes? What is the end result perceived by the user?*

No breaking changes, this is the first step in refactoring the code so it unlocks the two destination team EPICs to be able to run in parallel

## Pre-merge Checklist
*Expand the relevant checklist and delete the others.*

<details><summary><strong>New Connector</strong></summary>

### Community member or Airbyter

- [ ] **Community member?** Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret`
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Connector version is set to `0.0.1`
    - [ ] `Dockerfile` has version `0.0.1`
- [ ] Documentation updated
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] `docs/integrations/<source or destination>/<name>.md` including changelog with an entry for the initial version. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
    - [ ] `docs/integrations/README.md`
    - [ ] `airbyte-integrations/builds.md`
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.com/contributing-to-airbyte/issues-and-pull-requests)

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing
- [ ] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)
- [ ] After the connector is published, connector added to connector index as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
- [ ] Seed specs have been re-generated by building the platform and committing the changes to the seed spec files, as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)

</details>

<details><summary><strong>Updating a connector</strong></summary>

### Community member or Airbyter

- [ ] Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret`
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Connector version has been incremented
    - [ ] Version has been bumped according to our [Semantic Versioning for Connectors](https://docs.airbyte.com/contributing-to-airbyte/#semantic-versioning-for-connectors) guidelines
    - [ ] `Dockerfile` has updated version
- [ ] Documentation updated
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] Changelog updated in `docs/integrations/<source or destination>/<name>.md` with an entry for the new version. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)

- [ ] PR name follows [PR naming conventions](https://docs.airbyte.com/contributing-to-airbyte/issues-and-pull-requests)

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing
- [ ] New Connector version released on Dockerhub and connector version bumped by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)

</details>

<details><summary><strong>Connector Generator</strong></summary>

- [ ] Issue acceptance criteria met
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.com/contributing-to-airbyte/issues-and-pull-requests)
- [ ] If adding a new generator, add it to the [list of scaffold modules being tested](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connector-templates/generator/build.gradle#L41)
- [ ] The generator test modules (all connectors with `-scaffold` in their name) have been updated with the latest scaffold by running `./gradlew :airbyte-integrations:connector-templates:generator:testScaffoldTemplates` then checking in your changes
- [ ] Documentation which references the generator is updated as needed

</details>

### CODE REVIEW

File: settings.gradle [hunk 1]
```diff
@@ -105,6 +105,7 @@ if (!System.getenv().containsKey("SUB_BUILD") || System.getenv().get("SUB_BUILD"
     include ':airbyte-integrations:bases:base-java'
     include ':airbyte-integrations:bases:base-java-s3'
     include ':airbyte-integrations:bases:base-normalization'
+    include ':airbyte-integrations:bases:bases-destination-jdbc' // needs to be lexicographically after base-java and base-normalization to avoid race condition
```
Comment (jbfbell) [inline settings.gradle:108]: This is fine and I think eases my concerns about this being a landmine for other teams (and future us) we should maybe at least add a backlog/icebox task to sort through this eventually.

**Review by jbfbell — APPROVED** [review]

File: settings.gradle (hunk 1 above)
Comment (ryankfu) [reply settings.gradle:108]: Filed ticket here to track https://github.com/airbytehq/airbyte/issues/25251

**Pull request merged**